node/sync/src/block_sync.rs

```diff
@@ -711,6 +730,18 @@ impl<N: Network> BlockSync<N> {
             !is_timeout && !is_obsolete
         });
 
+        // After the retain loop, handle the banning of peers
+        for peer_ip in peers_to_ban {
+            trace!("Banning peer {peer_ip} for timing out on block requests");
+            self.tcp.banned_peers().update_ip_ban(peer_ip.ip());
+
+            let tcp = self.tcp.clone();
+            tokio::spawn(async move {
+                tcp.disconnect(peer_ip).await;
+                trace!("Peer disconnected!");
```

Comment:
nit: we're already getting `DEBUG`-level information on the peer being disconnected [here](https://github.com/AleoNet/snarkOS/blob/b0b89c3d1c5c3c6690472801b9b283edeb46b4d3/node/tcp/src/tcp.rs#L298)